components/dialog/dialog-mixin.js

```diff
@@ -354,7 +359,7 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			this.dispatchEvent(new CustomEvent(
 				'd2l-dialog-open', { bubbles: true, composed: true }
 			));
-		});
+		}, 0);
```

Comment:
rAF doesn't give the contents time to render in Firefox specifically. This isn't the first time I've encountered situations where FF schedules microtasks differently from Chrome in a way that messes up timings. `setTimeout(, 0)` does do the trick though.